src/main/java/com/icthh/xm/ms/entity/service/impl/XmEntityServiceImpl.java

```diff
@@ -510,46 +516,50 @@ private void deleteNewLink(Link linkToDelete) {
      */
     @LogicExtensionPoint("Search")
     @Override
-    @Transactional(readOnly = true)
     @FindWithPermission("XMENTITY.SEARCH")
     @PrivilegeDescription("Privilege to search for the xmEntity corresponding to the query")
     public Page<XmEntity> search(String query, Pageable pageable, String privilegeKey) {
-        return xmEntityPermittedSearchRepository.search(query, pageable, XmEntity.class, privilegeKey);
+        return invokeRepositoryWithTransactionControl(() ->
+            xmEntityPermittedSearchRepository.search(query, pageable, XmEntity.class, privilegeKey)
+        );
     }
 
     @LogicExtensionPoint("SearchV2")
     @Override
-    @Transactional(readOnly = true)
     @FindWithPermission("XMENTITY.SEARCH")
     @PrivilegeDescription("Privilege to search for the xmEntity corresponding to the query")
     public Page<XmEntity> searchV2(SearchDto searchDto, String privilegeKey) {
-        return xmEntityPermittedSearchRepository.searchForPage(searchDto, privilegeKey);
+        return invokeRepositoryWithTransactionControl(() ->
+            xmEntityPermittedSearchRepository.searchForPage(searchDto, privilegeKey)
+        );
     }
 
     @LogicExtensionPoint(value = "SearchByTemplate", resolver = TemplateTypeKeyResolver.class)
     @Override
-    @Transactional(readOnly = true)
     @FindWithPermission("XMENTITY.SEARCH")
     @PrivilegeDescription("Privilege to search for the xmEntity corresponding to the query")
     public Page<XmEntity> search(String template,
                                  TemplateParamsHolder templateParamsHolder,
                                  Pageable pageable,
                                  String privilegeKey) {
         String query = getTemplateQuery(template, templateParamsHolder);
-        return xmEntityPermittedSearchRepository.search(query, pageable, XmEntity.class, privilegeKey);
+        return invokeRepositoryWithTransactionControl(() ->
```

Comment:
invokeRepositoryWithTransactionControl must be upper than method call to correct handle lep transaction

Reply:
implemented like that